src/vs/workbench/contrib/chat/browser/chatSessions.contribution.ts

```diff
@@ -318,6 +322,12 @@ export class ChatSessionsService extends Disposable implements IChatSessionsServ
 				stripPathStartingSeparator: true,
 			}
 		}));
+
+		const chatService = this._chatServiceLazy.value;
+		this._register(autorun(reader => {
+			const models = chatService.chatModels.read(reader);
+			this.registerModelProgressListener(models);
```

Comment:
I don't have the big picture, but it seems like a step backwards to do this in this service. I think _chatServiceLazy is trying to avoid a recursive dependency? Actually I don't really see how it helps since you still have to get a IChatService instance in the constructor of this service. And then if the end result of this is just to inform the list that the local sessions have changed, why shouldn't the local sessions provider do that?

Reply:
I was debugging this because I wasn't sure how it worked, and I think it works because our services are created lazily, and it would fail if the `IChatSessionsService` was referenced in the `IChatService` constructor

Reply:
Yeah I think the provider having `onDidChangeChatSessionItems` and not firing is strange, the other providers do fire them in other cases but the local chat session provider is very bare bones so it doesn't really need to update for anything beyond tracking progress.
The extensions used to have the full responsability for the rendering but we want to start migrating things off of that pattern to create some rendering standard, for now status, description and icons fall into that category.

Not sure what the best pattern is for cases like this where core owns part of the UI and the extension another part of it, I guess we need to restrict the API accordingly but how do we manage the updates based on the chat model progress in this case?

Reply:
I don't quite understand and don't have full context on the API, happy to dig into it with you when we're back. But in general, I think we need a well-defined extension API, and a well-defined internal API for the view to be built on.